lib/ecto/adapters/postgres/connection.ex

```diff
@@ -910,6 +910,24 @@ if Code.ensure_loaded?(Postgrex) do
       |> Enum.join(" ")
     end
 
+    defp intersperse([], _separator, _mapper), do: []
+    defp intersperse(list, separator, mapper) do
+      [^separator | list] = Enum.reduce(list, [], fn x, acc ->
+        [separator, mapper.(x) | acc]
+      end)
+      :lists.reverse(list)
```

Comment:
That's true. It might be hard to remove the extra separator in that case, though since it will be buried deep inside the list (if we put it on the front) or at the very end if we put it on the back.

I presume manual recursion is the best solution here, isn't it?